core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -71,12 +71,16 @@ public int compare(RecordPointerAndKeyPrefix r1, RecordPointerAndKeyPrefix r2) {
         return prefixComparisonResult;
       }
     }
+
+    public void free() {
+      recordComparator.free();
```

Comment:
Since `SortComparator ` does not extend `RecordComparator`, `@Override` may not be necessary.